watertap/core/plugins/tests/test_solvers.py

```diff
@@ -0,0 +1,148 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+import pytest
+import pyomo.environ as pyo
+
+from pyomo.common.errors import ApplicationError
+from idaes.core.util.scaling import (set_scaling_factor, get_scaling_factor,
+        constraints_with_scale_factor_generator, unscaled_constraints_generator)
+from idaes.core.util import get_solver
+from watertap.core.plugins.solvers import IpoptWaterTAP
+
+class TestIpoptWaterTAP:
+    @pytest.fixture(scope="class")
+    def m(self):
+        m = pyo.ConcreteModel()
+        m.a = pyo.Var(initialize=1)
+        m.b = b = pyo.Block()
+        b.a = pyo.Var([1,2])
+
+        m.c = pyo.Constraint(expr=(0, 1./(m.a**2) ,1))
+        b.c = pyo.Constraint([1,2], rule= lambda b,i: (i-1, b.a[i], i+2))
```

Comment:
Just because I'm not used to writing constraints this way, I want to make sure I understand-- is this saying first that 
0 <= b.a[1] <= 3
and then
1 <= b.a[2] <= 4 ?

Reply:
Yep! That's a compact syntax for a range constraint.